internal/logger/logger.go

```diff
@@ -1,3 +1,108 @@
 package logger
 
-// TODO: Implement logger and refactor all the log statements to use it
+import (
+	"fmt"
+	"log"
+	"log/slog"
+	"os"
+)
+
+var LogLevelString = getLogLevel()
+
+// RawLogger is the raw global logger object used for calls wrapped by the logging package
+var RawLogger = InitLogger(LogLevelString, "")
+
+// InitLogger initializes a cluster-id specific child logger
+func InitLogger(logLevelString string, clusterID string) *slog.Logger {
+	var logLevel slog.Level
+	var addSource bool
+	switch logLevelString {
+	case "debug":
+		logLevel = slog.LevelDebug
+		addSource = true
+	case "info":
+		logLevel = slog.LevelInfo
+	case "warn", "warning":
+		logLevel = slog.LevelWarn
+	case "error":
+		logLevel = slog.LevelError
+	default:
+		log.Fatalln("Invalid log level:", logLevelString)
+	}
+
+	// Create JSON handler with options
+	opts := &slog.HandlerOptions{
+		Level:     logLevel,
+		AddSource: addSource,
+	}
+	handler := slog.NewJSONHandler(os.Stdout, opts)
```

Comment:
We may want to support writing to file at some point, so maybe we don't hardcode `os.Stdout`. I believe you can take in a `io.Writer` and when you call `InitLogger` you can specify `os.Stdout`. 

We could maybe default to stdout, but give the user a way to change it via a Cobra/Viper config.

Reply:
Would it be worth defining a structure to hold logging options (ie - `logger.Options`) from the outset here?

For example:
```
type Options struct {
    level string
    out   io.Writer
    // Any additional options we want to expose in the future
    ...
} 
```

Then this function can be called with:

```
func InitLogger(opts Options) *slog.Logger {
```